### PR TITLE
Doing some light updating in prep for 2.0.

### PR DESCRIPTION
**Pull request by Sfshaza**
Staged: https://sz-webdev-1.firebaseapp.com/faq

Please review, @kwalrath.

### CODE REVIEW

File: src/faq.md [hunk 1]
```diff
@@ -81,24 +91,32 @@ option.
...
+Dart takes out a few features of JavaScript, such as prototypes and the
+global `object:`. This streamlines the VM, enables faster execution,
+and makes it easier to do code completion and refactoring.
+And Dart adds some goodies.  To name a few:
 
 * User-defined operator methods.  We like the lightweight, readable code
-these give for
-<a href="/articles/low-level-html/improving-the-dom">our DOM interface</a>.
+  these give for our DOM interface. For more information, see the
+  [dart:html](https://www.dartdocs.org/documentation/html/latest/) library.
```
Comment (kwalrath) [inline src/faq.md]: That link is to the package:html docs, not the dart:html docs.

File: src/faq.md [hunk 2]
```diff
@@ -123,8 +141,7 @@ only when necessary.
 
 #### Q. How does Dart compare with using the Closure compiler on JavaScript?
 
-The idea of optional type annotations is similar.  Dart's are nicer
-syntactically.
+The idea of type annotations is similar.  Dart's are nicer syntactically.
```
Comment (kwalrath) [inline src/faq.md]: Not sure if deleting "optional" is correct.

File: src/faq.md [hunk 3]
```diff
@@ -186,10 +203,9 @@ You can rely on the [Dart analyzer][dartanalyzer]
 to flag any syntax problems. Alternatively, you may
 consider porting one feature at a time to Dart and using the
 [JavaScript interoperability library][jsinterop] as the common middle
-ground. Be sure to watch our Google I/O 2012 talk <a
-href="http://www.youtube.com/watch?v=EvACKPBo_R8">Migrating Code from GWT to
-Dart</a>, but keep in mind that it predates our JavaScript interoperability
-library.
+ground. Be sure to watch [Dart-JavaScript
+interoperability](https://www.youtube.com/watch?v=aIonwL-8hdE),
```
Comment (kwalrath) [inline src/faq.md]: Move the comma to before the ], so it doesn't go to a new line.

File: src/faq.md [hunk 4]
```diff
@@ -186,10 +203,9 @@ You can rely on the [Dart analyzer][dartanalyzer]
...
+a talk from the Dart Developer Summit 2016).
```
Comment (kwalrath) [inline src/faq.md]: delete "the"?

**Comment by kwalrath** [comment]
Someone else should review these changes. @kevmoo? @wmleler?

**Comment by kwalrath** [comment]
Also, we should update the list of browsers before we publish this page. I believe we support Microsoft Edge; I'm not sure about IE.

File: src/_articles/low-level-html/improving-the-dom.md [hunk 5]
```diff
@@ -1,632 +0,0 @@
----
```
Comment (kevmoo) [inline src/_articles/low-level-html/improving-the-dom.md]: Many hits? 404s are fine – teach search engines the right lesson

File: src/faq.md (hunk 1 above)
Comment (kevmoo) [inline src/faq.md]: https://api.dartlang.org/stable/dart-html – w/ the right redirect magic

File: src/_articles/low-level-html/improving-the-dom.md (hunk 5 above)
Comment (Sfshaza) [inline src/_articles/low-level-html/improving-the-dom.md]: Interesting! I usually set up a redirect.

Comment (kevmoo) [inline src/_articles/low-level-html/improving-the-dom.md]: It's not a big deal either way – we just have a redirect to deal w/ forever

File: src/faq.md [hunk 6]
```diff
@@ -181,24 +194,24 @@ are pointing to a brighter future for web developers. You can read a
...
 to flag any syntax problems. Alternatively, you may
 consider porting one feature at a time to Dart and using the
 [JavaScript interoperability library][jsinterop] as the common middle
-ground. Be sure to watch our Google I/O 2012 talk <a
-href="http://www.youtube.com/watch?v=EvACKPBo_R8">Migrating Code from GWT to
-Dart</a>, but keep in mind that it predates our JavaScript interoperability
-library.
+ground. Be sure to watch [Dart-JavaScript
+interoperability,](https://www.youtube.com/watch?v=aIonwL-8hdE)
+a talk from Dart Developer Summit 2016).
```
Comment (kwalrath) [inline src/faq.md]: Delete the )

File: src/faq.md [hunk 7]
```diff
@@ -20,31 +19,37 @@ We should also update the supported browser list.]
...
-  * Internet Explorer, versions 10 and 11
-    * Dart v1.5 was the last release to support Internet Explorer 9.
-  * Firefox, latest version
   * Chrome, latest version
-  * Safari for desktop, latest version
-  * Safari for mobile, latest version
+  * Firefox, latest version
+  * Microsoft Edge, latest version
+  * Internet Explorer, latest version
+  * Safari, latest version
```
Comment (kwalrath) [inline src/faq.md]: We can move the "latest version" up out of the list ("We support the latest version of the following browsers:").

File: src/faq.md [hunk 8]
```diff
@@ -20,31 +19,37 @@ We should also update the supported browser list.]
...
 Although no production browsers can execute Dart code directly,
 all modern browsers can execute Dart code that's been compiled to JavaScript.
-For convenience while you're developing Dart code,
-you can use a version of Chromium (nicknamed [Dartium])
+For convenience while you're developing Dart code on Dart 1.x,
+you can use a version of Chromium (nicknamed [Dartium]) that
 has the Dart VM integrated into it.
 
+A new development compiler, [_dartdevc_](/tools/dartdevc)
+(also known as _DDC_) is available for web development as of Dart 1.24.
```
Comment (kwalrath) [inline src/faq.md]: Add a comma after the )

File: src/faq.md [hunk 9]
```diff
@@ -20,31 +19,37 @@ We should also update the supported browser list.]
...
+Also see the [dartdevc FAQ.](/tools/dartdevc/faq)
+
 #### Q. How do I debug an app once it has been compiled to JavaScript?
 
-Dart web apps are easiest to debug with an IDE, like WebStorm,
-and Dartium. However,
-the [debugging](/tools/dart2js#debugging)
-section of the dart2js documentation
+Dart web apps are easiest to debug with an IDE, like WebStorm.
+The [debugging](/tools/dart2js#debugging) section of the dart2js documentation
```
Comment (kwalrath) [inline src/faq.md]: I think Dartium is still the easiest way to debug.

Should we revert but add some sort of a standard comment here so we can remember to update this?

In general, it'd be good to add a searchable string everywhere you expect to have to update as 2.0 approaches.

File: src/faq.md [hunk 10]
```diff
@@ -78,27 +83,35 @@ option.
...
+  [dart:html](https://api.dartlang.org/stable/dart-html ) library.
 
 * Lightweight syntax for anonymous functions.  You use them a lot in
-web programming; now they look great.  And they come with correct
-binding of <code>this</code> and full block-level lexical scoping, no gotchas.
+  web programming; now they look great.  And they come with correct
+  binding of <code>this</code> and full block-level lexical scoping,
+  no gotchas.
+
+Dart 2.0, currently in development, differs from 1.x one important way.
```
Comment (kwalrath) [inline src/faq.md]: way. -> way:

File: src/_articles/low-level-html/improving-the-dom.md (hunk 5 above)
Comment (kwalrath) [inline src/_articles/low-level-html/improving-the-dom.md]: This page is #50 in popularity, overall, and the 3rd most popular article on webdev, out of 4. (First by a long shot is /articles/get-data/json-web-service, followed by /performance/event-loop, followed by improving-the-dom and embedding-in-html.)

I'm inclined to redirect if we can do so helpfully, but I don't have strong feelings about it. Maybe we could redirect to the dart:html API docs (https://api.dartlang.org/stable/dart-html)?

**Pull request merged**